src/App.tsx

```diff
@@ -1,10 +1,59 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React,
+{
+  useCallback, useEffect, useMemo, useState,
+} from 'react';
 import { UserWarning } from './UserWarning';
+import { getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { FilterBy } from './types/FilterBy';
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer/Footer';
+import { ErrorType } from './types/ErrorType';
+import { ErrorInfo } from './components/ErrorInfo/ErrorInfo';
 
-const USER_ID = 0;
+const USER_ID = 12034;
 
 export const App: React.FC = () => {
+  const [todosFromServer, setTodosFromServer] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState(FilterBy.All);
+  const [errorMessage, setErrorMessage] = useState<ErrorType | null>(null);
+
+  const displayError = (error: ErrorType) => {
+    setErrorMessage(error);
+
+    setTimeout(() => setErrorMessage(null), 3000);
+  };
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodosFromServer)
+      .catch(() => displayError(ErrorType.UnableToLoad));
+  }, []);
+
+  const filterTodos = useCallback(
+    (
+      todos: Todo[],
+    ) => {
+      switch (filterBy) {
+        case FilterBy.Active:
+          return todos.filter(todo => !todo.completed);
+
+        case FilterBy.Completed:
+          return todos.filter(todo => todo.completed);
+
+        default:
+          return todos;
+      }
```

Comment:
Filtering logic should be brought out as a separate helper outside of components to improve code readability